### PR TITLE
Adds maintenance tablet apps (Health analyzer is one now too)

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

**Sprites were generously made by** https://github.com/Tramzz

This was a branch I've been putting off for months because my hatred for tablet apps brought me to removing computer parts, tablets, and ntnet, however I've now returned (after a lot more procrastination).

Adds Maintenance tablet applications to the game, so far there's only three of them, for proof of concept;

1. The health analyzer app (the chem analyzer part is removed entirely) move away from being given to medical/geneticists/detectives/RD.

There is on exception, which is the CMO, who gets to have the application on their tablet roundstart. Maybe it could be given to their role disk as well?

2. A camera application:

image
image

3. MODsuit control application:

image
image

image

Maintenance applications stand out from normal tablet apps because they can't be downloaded off the App store, and instead can only be found in maintenance, with a one-use download, cloning the application from a disk to a computer, or vice versa, will delete the old one, meaning you can only have one application at once.

## Why It's Good For The Game

This is more as a proof of concept for maintenance applications, but I also think that the analyzer application wasn't really that good as an app, you should either use a health analyzer or the wand in front of medbay, you shouldn't just have one in your tablet at all times because it makes it lame if your analyzer is stolen.

## Changelog

:cl: JohnFulpWillard, sprites by Tramzz
add: Added Maintenance tablet applications, applications that can't be cloned or downloaded from the store, instead you can find one app in maintenance.
balance: The Analyzer tablet application is also a maintenance tablet application now.
/:cl:

### CODE REVIEW

**Comment by TheGreatValvatorez** [comment]
i think these should atleast be buyable somewhere but for a premium price some of these are legit handy like for atmos techs etc and it adds more stuff to spend money on but for an assistant this is a neat thing to find aswell

**Comment by Jackal-boop** [comment]
The analyzer app is really useful for detectives

**Comment by MTandi** [comment]
Why is it non-clonable? Is there any harm in people spreading disks with camera or health analyzer apps?

**Comment by TheGreatValvatorez** [comment]
cuz that devoids it of its uniquess though i wish there was a program that would make you pay x person for x file

**Comment by MTandi** [comment]
Making it a maintenance loot is already unique enough. On contrary, it's fun if people could share more apps with disks with each other - more communication opportunity, no harm, and it will make people use disks and apps more in general.

File: code/modules/modular_computers/file_system/programs/maintenance/phys_scanner.dm [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/datum/computer_file/program/maintenance/phys_scanner
+	filename = "phys_scanner"
+	filedesc = "Physical Scanner"
+	category = PROGRAM_CATEGORY_MISC
+	extended_desc = "This program allows the tablet to scan physical objects and display a data output."
```
[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply code/modules/modular_computers/file_system/programs/maintenance/phys_scanner.dm:5]: never was

**Comment by JohnFulpWillard** [comment]
> The chemical scanner tablet app is the only way to view an atom's reagents that isn't typically scannable, like food or pills. I would really like for that ability to persist somewhere.

could it somehow be like some chemical analyzer that science can also print because it would maybe be more interesting than pinning it onto the health anaylzer

though i dont think it would be worth doing in this PR anyway

**Comment by private-tristan** [comment]
> could it somehow be like some chemical analyzer that science can also print because it would maybe be more interesting than pinning it onto the health anaylzer

There is already a chemical analyzer in the game (added by fermichem iirc), unsure if it scans pills/patches because it doesn't spawn roundstart and needs to be printed at medlathe.

**Comment by JohnFulpWillard** [comment]
marking as draft pending another app because i feel starting with only one new unique one is lame

**Comment by Overuseddart** [comment]
Ooh, maybe they could be the virus sending apps in maints. Like how bout a cartridge which sends a virus that makes ads pop up on the PDA screen, or a virus that makes the messenger include some of the emojis you can type in random places when you try to direct message someone.

**Comment by Overuseddart** [comment]
Maint tablet that has the black market uplink with items that have a 20% uptick in price and a 20 second shipping delay

**Comment by Overuseddart** [comment]
Maint tablet that works like the old camera bug, use your pda on a camera to bug it and you can look on marked cameras through the PDA. Maximum of 6 cameras

**Comment by Overuseddart** [comment]
Maint tablet that allows your PDA to download 1 app that would normally be restricted to your PDA and ID. (Excluding AI ones)

**Comment by Overuseddart** [comment]
> marking as draft pending another app because i feel starting with only one new unique one is lame

Maint tablet that allows you to use anonymous messaging.

**Comment by Overuseddart** [comment]
Maint tablet that allows you to download a PAI onto your PDA, gaining special features 

1. Holochassis is limited to one screen away from you, but has the access of the ID in the PDA
2. You can use gigabytes from the PDA to give your PAI extra points, 1 Gigabyte = 5 PAI points 
3. PAI gets acess to messaging and apps on your PAI 
4. If you emmag the PDA while a PAI is inside it the PAI gets the ability for a temporary x-ray (lasts for 30 seconds, blinds the PAI for 10 seconds after, 2 minutes cooldown) And you can spend TC on the PAI through an app to unlock different functions. (Faster doorjack for 1 tc more holochassis range with a recall for 2tc ability to turn into a badass drone with no tc [you have to supply them with your own weapons] for 12 tc)

**Comment by Overuseddart** [comment]
> marking as draft pending another app because i feel starting with only one new unique one is lame

(Please let one of these suggestions be good)

**Comment by JohnFulpWillard** [comment]
> Maint tablet that works like the old camera bug, use your pda on a camera to bug it and you can look on marked cameras through the PDA

Current camera bug should still be able to do that

Your pAI one sounds cool on paper but we already can insert pAIs into PDAs at the moment, letting them use your messenger while inside of it sounds cool however.

Also my idea was an app that lets you externally control a modsuit's ui lol

**Comment by Overuseddart** [comment]
> > Maint tablet that works like the old camera bug, use your pda on a camera to bug it and you can look on marked cameras through the PDA
> 
> Current camera bug should still be able to do that
> 
> Your pAI one sounds cool on paper but we already can insert pAIs into PDAs at the moment, letting them use your messenger while inside of it sounds cool however.
> 
> Also my idea was an app that lets you externally control a modsuit's ui lol

That could be really useful. Maybe for the pAI into PDA we can add flavor text that says you are now able to download the pAI directly into your systems Cloud and storage, and the app “Jailbreaks” the tablet allowing you to allow the pAI to use your messages?

**Comment by Overuseddart** [comment]
Hey, is this still being worked on?

**Comment by JohnFulpWillard** [comment]
Added MODsuit control app

image

Now all that's needed is the ability to deploy/undeploy parts from the UI just like how you can activate it through it, and MODsuits would be good finally

**Comment by optimumtact** [comment]
>The chemical scanner tablet app is the only way to view an atom's reagents that isn't typically scannable, like food or pills. I would really like for that ability to persist somewhere.

Your'e not even supposed to be able to do this.

**Comment by JohnFulpWillard** [comment]
This PR is currently just held up by lack of sprites, I'm not a spriter so I think it's jover

**Comment by NamelessFairy** [comment]
> This PR is currently just held up by lack of sprites, I'm not a spriter so I think it's jover

What sprites does it need?

**Pull request merged**